ros2service/test/test_echo.py

```diff
@@ -93,12 +95,15 @@ def generate_test_description(rmw_implementation):
         os.path.dirname(__file__), 'fixtures', 'introspectable.py'
     )
     additional_env = get_rmw_additional_env(rmw_implementation)
+    set_env_actions = [SetEnvironmentVariable(k, v) for k, v in additional_env.items()]
     return LaunchDescription([
         # Always restart daemon to isolate tests.
         ExecuteProcess(
             cmd=['ros2', 'daemon', 'stop'],
             name='daemon-stop',
             on_exit=[
+                *set_env_actions,
+                EnableRmwIsolation(),
```

Comment:
the following `ros2 daemon` process and test fixture node will be created with multicast based scout discovery protocol because it still uses `additional_env`? that fails the actual test command which uses zenoh router, because it could not find the test fixture node via zenoh router.

Reply:
The multicast configuration was dropped from `additional_env` (see changes to `helpers.py`), so only `RMW_IMPLEMENTATION` and `RUST_LOG` are getting set. The isolation (and Zenoh configuration) happens with `EnableRmwIsolation`.

The `additional_env=additional_env` below is redundant with `*set_env_actions` here and should indeed be removed, though.